extensions/markdown-language-features/src/test/markdownLink.test.ts

```diff
@@ -0,0 +1,79 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import * as vscode from 'vscode';
+import { checkSmartPaste, createLinkSnippet } from '../languageFeatures/copyFiles/shared';
+import 'mocha';
+import { validateLink } from '../languageFeatures/copyFiles/copyPasteLinks';
+
+suite('uriToWorkspaceEdit', () => {
+
+	// end to end test of checkSmartPaste & createLinkSnippet
+	// check multicursor (end to end)
+
+	// check paste with selection
+	// check paste over markdown link
+	// check paste over markdown image
+	// check paste within markdown link
+	// check paste with open bracket
+
+	// check image link
+
+	////////////////////////////////// validateLink tests //////////////////////////
+	test('Markdown pasting should occur for a valid link.', () => {
+		const isLink = validateLink('https://www.microsoft.com');
+		assert.strictEqual(isLink, true);
+	});
+
+	test('Markdown pasting should not occur for a link following by text.', () => {
+		const isLink = validateLink('https://www.microsoft.com hello world!');
+		assert.strictEqual(isLink, false);
+	});
+
+	test('Markdown pasting should not occur for multiple links being pasted.', () => {
+		const isLink = validateLink('https://www.microsoft.com\r\nhttps://www.microsoft.com\r\nhttps://www.microsoft.com\r\nhttps://www.microsoft.com');
+		assert.strictEqual(isLink, false);
+	});
+
```

Comment:
A few other cases to consider:

- Text that would have passed when using `new Url(...)`
- Text with whitespace
- Valid uris with unsupported schemes